From 864ecf16d010628abed0242db2d4fa2effb9965d Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@gmail.com>
Date: Fri, 21 Nov 2014 15:34:28 +0100
Subject: Remove fopencookie hack.

* mmu_man added these prototypes in Haiku so this now breaks the build.

diff --git a/ext/solv_xfopen.c b/ext/solv_xfopen.c
index 63fec23..a9a6e40 100644
--- a/ext/solv_xfopen.c
+++ b/ext/solv_xfopen.c
@@ -17,23 +17,6 @@
 #include "util.h"
 
 
-/* Evil hack for Haiku: fopencookie() is implemented internally, but not
-   exported by a header. */
-#ifdef __HAIKU__
-
-typedef struct {
-	ssize_t (*read)(void*, char*, size_t);
-	ssize_t (*write)(void*, const char*, size_t);
-	int (*seek)(off_t*, int);
-	int (*close)(void*);
-} cookie_io_functions_t;
-
-
-FILE *fopencookie(void*, const char*, cookie_io_functions_t);
-
-#endif /* __HAIKU__ */
-
-
 static FILE *cookieopen(void *cookie, const char *mode,
 	ssize_t (*cread)(void *, char *, size_t), 
 	ssize_t (*cwrite)(void *, const char *, size_t), 
-- 
1.8.3.4

